reset extension data error message

Mathieu-Deharbe opened the pull request:
lost during the generic computation classes update.

This PR is a very simple correction but it might be done in a more elegant exception based way. See the following work currently done : 
https://github.com/gridsuite/shortcircuit-server/pull/95
https://github.com/powsybl/powsybl-ws-commons/pull/58

possible idea : A throw launched in the sendResultMessage function if the result is invalid. This throw will be catched by the standard functions.

etiennehomer commented on src/main/java/org/gridsuite/shortcircuit/server/service/ShortCircuitWorkerService.java:
```diff
@@ -72,8 +72,13 @@ protected void sendResultMessage(AbstractResultContext<ShortCircuitRunContext> r
+        if (!result.getFaultResults().isEmpty() && resultContext.getRunContext().getBusId() == null &&
+                result.getFaultResults().stream().map(FaultResult::getStatus).allMatch(FaultResult.Status.NO_SHORT_CIRCUIT_DATA::equals)) {
+            throw new ShortCircuitException(MISSING_EXTENSION_DATA, "Missing short-circuit extension data");
+        } else {
```
you can remove the "else"

Mathieu-Deharbe replied:
Done.

etiennehomer reviewed (commented):
It would be nice to have a small unit test on this case

Mathieu-Deharbe commented:
> It would be nice to have a small unit test on this case

Indeed, that would be dumb to encounter the same problem agin. There was already a test but dysfunctional. I corrected it and added a specific network for it.
